pyro/infer/advi.py

```diff
@@ -81,7 +82,10 @@ def model(self, *args, **kwargs):
         with poutine.scale("advi_scope", 0.0):
             self.sample_latent(*args, **kwargs)
         # actual model sample statements shouldn't be zeroed out
-        return self.base_model(*args, **kwargs)
+        base_trace = poutine.trace(self.base_model).get_trace(*args, **kwargs)
```

Comment:
if you're gonna do this might you not as well block all the other sites?

Reply:
shouldn't it error if there exist any _subsample sites?

Reply:
> if you're gonna do this might you not as well block all the other sites?

What other sites? `base_trace` should contain exactly the same sites as the `prototype_trace`.

> shouldn't it error if there exist any _subsample sites?

Why is that? I have pruned the subsample sites exactly as you did when constructing the `prototype_trace`. Indeed when I failed to prune those sites, the tests failed.

Reply:
well doesn't a subsample hint at being outside of the workable model class?

Reply:
Can you clarify whether you believe this check to be too lax or too strict? I'm having a hard time understanding what you're requesting.

Reply:
i'm fine with the check (assuming it will later be gated).

i'm just suggesting that certain kinds of subsampling (e.g. subsampling local random variables) will break this. maybe a check of some sort is possible. maybe there is no need for a check and we should just add some more textual warnings to the docstring

Reply:
Thanks, that makes sense. I think we can even simply check once in the `prototype_trace`. I'm not sure exactly what the check should be, so maybe we should chat and add it to a follow-up PR.

For example, I'm not sure whether it's ok to subsample observations, since we might allow the `latent` to depend on `*args, **kwargs` through which a subsample might be passed.